### ISSUE TITLE
ModalPane allows focus to escape under it

### ISSUE THREAD

**Issue by iajn** [mkpaz/atlantafx#63]
Noticed while playing around in the sampler app, when a ModalPane is visible its possible to shift the focus using tab somewhere under it and then activate buttons using spacebar. I'd imagine focus should shift only inside the dialog. Not sure if this can be limited with some existing property?

In the sampler app, its possible to lock the UI by opening the persistent ModalPane, shifting the focus to some other button and opening a different dialog that doesn't have a close button.

**Comment by mkpaz** [comment]
Yes, the focus should be cleared somehow before moving the front layer to the back. The only issue is that JavaFX does not provide an API for this.

**Comment by mkpaz** [comment]
I dug a little deeper into this issue and don't see anything that can be done without platform support. When one press TAB, the control that has the focus (e.g. TextField) [consumes the KeyEvent](https://github.com/openjdk/jfx/blob/master/modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/behavior/TextInputControlBehavior.java#L172) so that it doesn't bubble up to its parent (the ModalPane) and calls a [TraverseEngine](https://github.com/openjdk/jfx/blob/master/modules/javafx.graphics/src/main/java/com/sun/javafx/scene/traversal/TraversalEngine.java#L45). The algorithm passes the focus to the [nearest](https://github.com/openjdk/jfx/blob/master/modules/javafx.graphics/src/main/java/com/sun/javafx/scene/traversal/TabOrderHelper.java#L36) element in the scene graph, based on distance. Like almost everything in JavaFX, it's a private API, so no tweaks allowed.

Workaround^ on the `show` event set `focusTraversable` to false for _every_ node in the current scene except the ModalPane and its children. Set it back to true on hide.

It's quite messy because we don't know the depth of the scene graph beforehand, so I wouldn't implement this in the control itself.

**Comment by mkpaz** [comment]
Perhaps something can be done after the focus delegation API is implemented in JavaFX.

https://github.com/openjdk/jfx/pull/1632
